internalshared/configutil/datadog_sink.go

```diff
@@ -0,0 +1,111 @@
+package configutil
+
+import (
+	"github.com/armon/go-metrics"
+	"github.com/armon/go-metrics/datadog"
+	"github.com/mitchellh/cli"
+)
+
+type DatadogSink struct {
+	tags              []string
+	addr              string
+	hostName          string
+	propagateHostname bool
+	sink              *datadog.DogStatsdSink
+	logger            cli.Ui
+}
+
+func NewDatadogSink(addr string, hostName string, logger cli.Ui) (*DatadogSink, error) {
+	return &DatadogSink{
+		addr:              addr,
+		hostName:          hostName,
+		propagateHostname: false,
+		logger:            logger,
+	}, nil
+}
```

Comment:
Since there's no way for this to error, what are your thoughts on removing the returned nil error value here and also removing the error check in `internalshared/configutil/telemetry.go` where this is called?

Reply:
Fixed